Source/Diagnostics/WarpXIO.cpp

```diff
@@ -522,18 +527,24 @@ WarpX::WritePlotFile () const
     }
 
 #ifdef WARPX_USE_OPENPMD
+    m_OpenPMDPlotWriter->SetStep(istep[0]);
+    if (dump_openpmd)
+      m_OpenPMDPlotWriter->WriteOpenPMDFields(varnames,
```

Comment:
I think we should move this out of this `WritePlotFile()` routine.
The `varnames` logic mages this hard to generalize to RZ later on and we do not gain from the packing and re-construction needed for plotfiles.

Reply:
this is. where. openpmd field. writer was.... I am leave it as is for now as. I am not sure. how RZ. works.

Reply:
The was the location of the field data storage code.  Will leave it as is until further clarification.